lib/streamlit/__init__.py

```diff
@@ -194,347 +184,30 @@ def _update_logger() -> None:
-        return
-    ctx.query_string = _parse.urlencode(query_params, doseq=True)
-    msg = _ForwardMsg_pb2.ForwardMsg()
-    msg.page_info_changed.query_string = ctx.query_string
-    ctx.enqueue(msg)
-
-
-@_contextlib.contextmanager
-def spinner(text: str = "In progress...") -> Iterator[None]:
-    """Temporarily displays a message while executing a block of code.
-
-    Parameters
-    ----------
-    text : str
-        A message to display while executing that block
-
-    Example
-    -------
-
-    >>> with st.spinner('Wait for it...'):
-    >>>     time.sleep(5)
-    >>> st.success('Done!')
-
-    """
-    import streamlit.runtime.legacy_caching.caching as legacy_caching
-    import streamlit.runtime.caching as caching
-    from streamlit.proto.Spinner_pb2 import Spinner as SpinnerProto
-    from streamlit.string_util import clean_text
-
-    # @st.cache optionally uses spinner for long-running computations.
-    # Normally, streamlit warns the user when they call st functions
-    # from within an @st.cache'd function. But we do *not* want to show
-    # these warnings for spinner's message, so we create and mutate this
-    # message delta within the "suppress_cached_st_function_warning"
-    # context.
-    with legacy_caching.suppress_cached_st_function_warning():
-        with caching.suppress_cached_st_function_warning():
-            message = empty()
-
-    try:
-        # Set the message 0.1 seconds in the future to avoid annoying
-        # flickering if this spinner runs too quickly.
-        DELAY_SECS = 0.1
-        display_message = True
-        display_message_lock = _threading.Lock()
-
-        def set_message():
-            with display_message_lock:
-                if display_message:
-                    with legacy_caching.suppress_cached_st_function_warning():
-                        with caching.suppress_cached_st_function_warning():
-                            spinner_proto = SpinnerProto()
-                            spinner_proto.text = clean_text(text)
-                            message._enqueue("spinner", spinner_proto)
-
-        _add_script_run_ctx(_threading.Timer(DELAY_SECS, set_message)).start()
-
-        # Yield control back to the context.
-        yield
-    finally:
-        if display_message_lock:
-            with display_message_lock:
-                display_message = False
-        with legacy_caching.suppress_cached_st_function_warning():
-            with caching.suppress_cached_st_function_warning():
-                message.empty()
+# Experimental APIs
+experimental_user = _UserInfoProxy()
+experimental_singleton = _singleton
+experimental_memo = _memo
+experimental_get_query_params = _gather_metrics(_get_query_params)
+experimental_set_query_params = _gather_metrics(_set_query_params)
+experimental_show = _gather_metrics(_show)
+experimental_rerun = _rerun
 
 
 @_gather_metrics
-def _transparent_write(*args: Any) -> Any:
+def _transparent_write(*args: _Any) -> _Any:
```

Comment:
nit: Wouldn't it be possible to move `_transparent_write` also to the `write` module as well and set it via something like `_transparent_write = main.transparent_write`? It seems to be the last remaining command like function definition in `__init__.py`.

Reply:
Yeah - `_transparent_write` is part of the `magic` module, and arguably belongs in there. It's the most fraught to move, though, so I prefer it being in its own PR.